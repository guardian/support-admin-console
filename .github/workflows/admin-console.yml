name: Build admin-console

on:
  pull_request:
    branches:
      - "**"
  push:
    branches:
      - main

jobs:
  admin-console:
    if: >-
      (github.event.pull_request.head.repo.owner.login == 'guardian' ||
        github.event_name == 'push')
    # Required by actions-assume-aws-role
    permissions:
      id-token: write
      contents: read

    name: admin-console
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repo
        uses: actions/checkout@v3

      - name: Use Node.js
        uses: actions/setup-node@v3
        with:
          node-version-file: '.nvmrc'

      - name: Test + build
        run: |
          npm install
          npm run lint
          npm run test
          npm run build

      - name: cdk
        working-directory: ./cdk
        run: |
          npm install -g yarn
          ./script/ci

      - name: Configure AWS credentials
        uses: aws-actions/configure-aws-credentials@v2
        with:
          role-to-assume: ${{ secrets.GU_RIFF_RAFF_ROLE_ARN }}
          aws-region: eu-west-1

      - name: Setup Java 11
        uses: actions/setup-java@v3
        with:
          distribution: "corretto"
          java-version: "11"
      - uses: actions/cache@v3
        with:
          path: |
            ~/.ivy2/cache
            ~/.sbt
            ~/.coursier
          key: sbt

      - name: riffraff
        run: |
          export LAST_TEAMCITY_BUILD=2000
          export GITHUB_RUN_NUMBER=$(( $GITHUB_RUN_NUMBER + $LAST_TEAMCITY_BUILD ))

          sbt clean riffRaffUpload
