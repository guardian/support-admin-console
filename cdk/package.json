{
  "name": "cdk",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "tsc",
    "test": "jest",
    "format": "prettier --write \"{lib,bin}/**/*.ts\"",
    "lint": "eslint lib/** bin/** --ext .ts --no-error-on-unmatched-pattern",
    "synth": "cdk synth --path-metadata false --version-reporting false",
    "diff": "cdk diff --path-metadata false --version-reporting false"
  },
  "devDependencies": {
    "@guardian/cdk": "61.3.2",
    "@guardian/eslint-config-typescript": "10.0.1",
    "@types/jest": "^27.4.1",
    "@types/node": "18.18.0",
    "aws-cdk": "2.1005.0",
    "aws-cdk-lib": "2.185.0",
    "constructs": "10.3.0",
    "eslint": "8.56.0",
    "jest": "29.7.0",
    "prettier": "^2.6.2",
    "source-map-support": "^0.5.20",
    "ts-jest": "29.1.2",
    "tslib": "2.6.2",
    "typescript": "5.3.3"
  },
  "resolutions": {
    "json5": "^2.2.2"
  }
}
