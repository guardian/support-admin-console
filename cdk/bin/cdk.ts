import 'source-map-support/register';
import { App } from '@aws-cdk/core';
import { AdminConsole } from '../lib/admin-console';

const app = new App();
new AdminConsole(app, 'AdminConsole-CODE', { stack: 'support', stage: 'CODE' });
new AdminConsole(app, 'AdminConsole-PROD', { stack: 'support', stage: 'PROD' });
