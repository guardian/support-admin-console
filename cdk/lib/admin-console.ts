import { GuEc2App } from '@guardian/cdk';
import { AccessScope } from '@guardian/cdk/lib/constants';
import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { GuStack } from '@guardian/cdk/lib/constructs/core';
import { GuCname } from '@guardian/cdk/lib/constructs/dns';
import {
  GuAllowPolicy,
  GuDynamoDBReadPolicy,
  GuDynamoDBWritePolicy,
  GuGetS3ObjectsPolicy,
  GuPutS3ObjectsPolicy,
} from '@guardian/cdk/lib/constructs/iam';
import type { App, CfnElement } from 'aws-cdk-lib';
import { Duration, RemovalPolicy } from 'aws-cdk-lib';
import { AttributeType, ProjectionType, Table } from 'aws-cdk-lib/aws-dynamodb';
import { InstanceClass, InstanceSize, InstanceType } from 'aws-cdk-lib/aws-ec2';

export interface AdminConsoleProps extends GuStackProps {
  domainName: string;
}

export class AdminConsole extends GuStack {
  // Build a dynamodb table to store tests for the given channel
  buildTestsTable(channel: string): Table {
    const id = `${channel[0].toUpperCase() + channel.slice(1)}TestsDynamoTable`;

    const table = new Table(this, id, {
      tableName: `${channel}-tests-${this.stage}`,
      removalPolicy: RemovalPolicy.RETAIN,
      readCapacity: 4,
      writeCapacity: 4,
      partitionKey: {
        name: 'name',
        type: AttributeType.STRING,
      },
    });

    // Add an index for querying by status (LIVE/DRAFT/ARCHIVED)
    table.addGlobalSecondaryIndex({
      indexName: 'status-index',
      partitionKey: {
        name: 'status',
        type: AttributeType.STRING,
      },
      readCapacity: 4,
      writeCapacity: 4,
      projectionType: ProjectionType.ALL,
    });

    // Give it a better name
    const defaultChild = table.node.defaultChild as unknown as CfnElement;
    defaultChild.overrideLogicalId(id);

    return table;
  }

  buildDynamoPolicies(tables: Table[]): GuAllowPolicy[] {
    const dynamoReadPolicies = tables.map(
      (table) =>
        new GuDynamoDBReadPolicy(this, `DynamoRead-${table.node.id}`, {
          tableName: table.tableName,
        }),
    );
    const dynamoWritePolicies = tables.map(
      (table) =>
        new GuDynamoDBWritePolicy(this, `DynamoWrite-${table.node.id}`, {
          tableName: table.tableName,
        }),
    );

    return [...dynamoReadPolicies, ...dynamoWritePolicies];
  }

  constructor(scope: App, id: string, props: AdminConsoleProps) {
    super(scope, id, props);

    const { domainName } = props;

    const app = 'admin-console';

    const dynamoTables = [
      this.buildTestsTable('header'),
      this.buildTestsTable('epic'),
      this.buildTestsTable('banner'),
    ];
    const dynamoPolicies = this.buildDynamoPolicies(dynamoTables);

    const userData = `#!/bin/bash -ev
    aws --region ${this.region} s3 cp s3://membership-dist/${this.stack}/${this.stage}/${app}/support-admin-console_1.0-SNAPSHOT_all.deb /tmp
    dpkg -i /tmp/support-admin-console_1.0-SNAPSHOT_all.deb
    /opt/cloudwatch-logs/configure-logs application ${this.stack} ${this.stage} ${app} /var/log/support-admin-console/application.log`;

    const policies = [
      new GuAllowPolicy(this, 'Cloudwatch', {
        actions: ['logs:CreateLogGroup', 'logs:CreateLogStream', 'logs:PutLogEvents'],
        resources: ['arn:aws:logs:*:*:*'],
      }),
      new GuAllowPolicy(this, 'SSMGet', {
        actions: ['ssm:GetParametersByPath'],
        resources: [`arn:aws:ssm:${this.region}:${this.account}:parameter/${app}/${this.stage}`],
      }),
      new GuGetS3ObjectsPolicy(this, 'SettingsBucketGet', {
        bucketName: 'support-admin-console',
        paths: [`${this.stage}/*`, 'google-auth-service-account-certificate.json'],
      }),
      new GuPutS3ObjectsPolicy(this, 'SettingsBucketPut', {
        bucketName: 'support-admin-console',
        paths: [`${this.stage}/*`],
      }),
      new GuAllowPolicy(this, 'PublicSettingsBucketPut', {
        actions: ['s3:PutObject', 's3:PutObjectAcl'],
        resources: [
          `arn:aws:s3:::gu-contributions-public/epic/${this.stage}/*`,
          `arn:aws:s3:::gu-contributions-public/banner/${this.stage}/*`,
          `arn:aws:s3:::gu-contributions-public/header/${this.stage}/*`,
        ],
      }),
      ...dynamoPolicies,
    ];

    const ec2App = new GuEc2App(this, {
      applicationPort: 9000,
      app,
      access: { scope: AccessScope.PUBLIC },
      certificateProps: {
        domainName,
      },
      monitoringConfiguration: {
        noMonitoring: true,
      },
      userData,
      roleConfiguration: {
        additionalPolicies: policies,
      },
      scaling: { minimumInstances: 1, maximumInstances: 2 },
      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MICRO),
      withoutImdsv2: true,
    });

    new GuCname(this, 'cname', {
      app,
      domainName,
      ttl: Duration.minutes(60),
      resourceRecord: ec2App.loadBalancer.loadBalancerDnsName,
    });
  }
}
