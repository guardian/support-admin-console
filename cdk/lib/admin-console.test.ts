import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { prodProps } from '../bin/cdk';
import { AdminConsole } from './admin-console';

describe('The AdminConsole stack', () => {
  it('matches the snapshot', () => {
    const app = new App();
    const stack = new AdminConsole(app, 'AdminConsole', prodProps);
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
