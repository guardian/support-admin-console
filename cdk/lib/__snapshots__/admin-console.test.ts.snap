// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AdminConsole stack matches the snapshot 1`] = `
Object {
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "support-admin-console",
  "Parameters": Object {
    "AMI": Object {
      "Default": "ami-0eb88168",
      "Description": "AMI ID (may be replaced by RiffRaff for latest baked AMI)",
      "Type": "String",
    },
    "App": Object {
      "Default": "admin-console",
      "Description": "Applied directly as a tag",
      "Type": "String",
    },
    "BastionSecurityGroup": Object {
      "Description": "Bastion's security group for SSH",
      "Type": "String",
    },
    "CertificateARN": Object {
      "Description": "SSL certificate ARN",
      "Type": "String",
    },
    "KingsPlaceIP": Object {
      "Default": "77.91.248.0/21",
      "Description": "Kings Place IP range",
      "Type": "String",
    },
    "PrivateVpcSubnets": Object {
      "Description": "Private subnets to use for EC2 instances",
      "Type": "List<AWS::EC2::Subnet::Id>",
    },
    "PublicVpcSubnets": Object {
      "Description": "Public subnets to use for the ELB",
      "Type": "List<AWS::EC2::Subnet::Id>",
    },
    "Stack": Object {
      "Description": "Applied directly as a tag",
      "Type": "String",
    },
    "Stage": Object {
      "Description": "Applied directly as a tag",
      "Type": "String",
    },
    "VpcId": Object {
      "Default": "vpc-e6e00183",
      "Description": "VpcId of your existing Virtual Private Cloud (VPC)",
      "Type": "String",
    },
  },
  "Resources": Object {
    "AppRole": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sts:AssumeRole",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "Service": Array [
                  "ec2.amazonaws.com",
                ],
              },
            },
          ],
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:aws:iam::\${AWS::AccountId}:policy/guardian-ec2-role-for-ssm",
          },
        ],
        "Path": "/",
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "logs:CreateLogGroup",
                    "logs:CreateLogStream",
                    "logs:PutLogEvents",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::GetAtt": "LogGroup.Arn",
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "CloudwatchLogs",
          },
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "ssm:GetParametersByPath",
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Sub": "arn:aws:ssm:\${AWS::Region}:\${AWS::AccountId}:parameter/\${App}/\${Stage}",
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "ConfigLoad",
          },
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "s3:*",
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Sub": "arn:aws:s3:::support-admin-console/\${Stage}/*",
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "SettingsBucket",
          },
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "s3:*",
                  "Effect": "Allow",
                  "Resource": Array [
                    Object {
                      "Fn::Sub": "arn:aws:s3:::gu-contributions-public/epic/\${Stage}/*",
                    },
                    Object {
                      "Fn::Sub": "arn:aws:s3:::gu-contributions-public/header/\${Stage}/*",
                    },
                    Object {
                      "Fn::Sub": "arn:aws:s3:::gu-contributions-public/banner/\${Stage}/*",
                    },
                  ],
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "PublicSettingsBucket",
          },
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "s3:GetObject",
                  "Effect": "Allow",
                  "Resource": "arn:aws:s3:::support-admin-console/google-auth-service-account-certificate.json",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "GoogleServiceAccountCertificate",
          },
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "autoscaling:DescribeAutoScalingInstances",
                    "autoscaling:DescribeAutoScalingGroups",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "DescribeAutoScaling",
          },
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "ec2:DescribeTags",
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "DescribeTags",
          },
          Object {
            "PolicyDocument": Object {
              "Statement": Object {
                "Action": "s3:GetObject",
                "Effect": "Allow",
                "Resource": Array [
                  "arn:aws:s3::*:membership-dist/*",
                ],
              },
            },
            "PolicyName": "s3Deploy",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "40.0.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-admin-console",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "AutoScalingGroup": Object {
      "Properties": Object {
        "HealthCheckGracePeriod": 300,
        "HealthCheckType": "ELB",
        "LaunchConfigurationName": Object {
          "Ref": "LaunchConfig",
        },
        "MaxSize": "2",
        "MinSize": "1",
        "Tags": Array [
          Object {
            "Key": "App",
            "PropagateAtLaunch": true,
            "Value": Object {
              "Ref": "App",
            },
          },
          Object {
            "Key": "gu:cdk:version",
            "PropagateAtLaunch": true,
            "Value": "40.0.1",
          },
          Object {
            "Key": "gu:repo",
            "PropagateAtLaunch": true,
            "Value": "guardian/support-admin-console",
          },
          Object {
            "Key": "Name",
            "PropagateAtLaunch": true,
            "Value": Object {
              "Fn::Join": Array [
                "-",
                Array [
                  Object {
                    "Ref": "Stack",
                  },
                  Object {
                    "Ref": "Stage",
                  },
                  Object {
                    "Ref": "App",
                  },
                ],
              ],
            },
          },
          Object {
            "Key": "Stack",
            "PropagateAtLaunch": true,
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
        ],
        "TargetGroupARNs": Array [
          Object {
            "Ref": "TargetGroup",
          },
        ],
        "VPCZoneIdentifier": Object {
          "Ref": "PrivateVpcSubnets",
        },
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
    },
    "ElasticLoadBalancer": Object {
      "Properties": Object {
        "Name": Object {
          "Fn::Sub": "\${Stack}-\${Stage}-\${App}",
        },
        "SecurityGroups": Array [
          Object {
            "Ref": "LoadBalancerSecurityGroup",
          },
        ],
        "Subnets": Object {
          "Ref": "PublicVpcSubnets",
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": Object {
              "Ref": "App",
            },
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "40.0.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-admin-console",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "InstanceProfile": Object {
      "Properties": Object {
        "Path": "/",
        "Roles": Array [
          Object {
            "Ref": "AppRole",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "InstanceSecurityGroup": Object {
      "Properties": Object {
        "GroupDescription": "Open up SSH access and enable HTTP access on the configured port",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "SecurityGroupIngress": Array [
          Object {
            "FromPort": 22,
            "IpProtocol": "tcp",
            "SourceSecurityGroupId": Object {
              "Ref": "BastionSecurityGroup",
            },
            "ToPort": 22,
          },
          Object {
            "CidrIp": Object {
              "Ref": "KingsPlaceIP",
            },
            "FromPort": 9000,
            "IpProtocol": "tcp",
            "ToPort": 9000,
          },
          Object {
            "FromPort": 9000,
            "IpProtocol": "tcp",
            "SourceSecurityGroupId": Object {
              "Ref": "LoadBalancerSecurityGroup",
            },
            "ToPort": 9000,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "40.0.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-admin-console",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LaunchConfig": Object {
      "Properties": Object {
        "AssociatePublicIpAddress": false,
        "IamInstanceProfile": Object {
          "Ref": "InstanceProfile",
        },
        "ImageId": Object {
          "Ref": "AMI",
        },
        "InstanceType": "t4g.micro",
        "SecurityGroups": Array [
          Object {
            "Ref": "InstanceSecurityGroup",
          },
          Object {
            "Ref": "WazuhSecurityGroup",
          },
        ],
        "UserData": Object {
          "Fn::Base64": Object {
            "Fn::Sub": "#!/bin/bash -ev
aws --region \${AWS::Region} s3 cp s3://membership-dist/\${Stack}/\${Stage}/\${App}/support-admin-console_1.0-SNAPSHOT_all.deb /tmp
dpkg -i /tmp/support-admin-console_1.0-SNAPSHOT_all.deb
/opt/cloudwatch-logs/configure-logs application \${Stack} \${Stage} \${App} /var/log/support-admin-console/application.log
",
          },
        },
      },
      "Type": "AWS::AutoScaling::LaunchConfiguration",
    },
    "LoadBalancerListener": Object {
      "Properties": Object {
        "Certificates": Array [
          Object {
            "CertificateArn": Object {
              "Ref": "CertificateARN",
            },
          },
        ],
        "DefaultActions": Array [
          Object {
            "TargetGroupArn": Object {
              "Ref": "TargetGroup",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Ref": "ElasticLoadBalancer",
        },
        "Port": 443,
        "Protocol": "HTTPS",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "LoadBalancerSecurityGroup": Object {
      "Properties": Object {
        "GroupDescription": "Permit incoming HTTPS access on port 443, egress to port 9000",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "FromPort": 9000,
            "IpProtocol": "tcp",
            "ToPort": 9000,
          },
        ],
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "40.0.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-admin-console",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LogGroup": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Sub": "\${Stack}-\${App}-\${Stage}",
        },
        "RetentionInDays": 14,
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "40.0.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-admin-console",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::Logs::LogGroup",
    },
    "TargetGroup": Object {
      "DependsOn": Array [
        "ElasticLoadBalancer",
      ],
      "Properties": Object {
        "HealthCheckIntervalSeconds": 10,
        "HealthCheckPath": "/healthcheck",
        "HealthCheckPort": "9000",
        "HealthCheckProtocol": "HTTP",
        "HealthCheckTimeoutSeconds": 5,
        "HealthyThresholdCount": 2,
        "Name": Object {
          "Fn::Sub": "\${Stack}-\${Stage}-\${App}",
        },
        "Port": 9000,
        "Protocol": "HTTP",
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "40.0.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-admin-console",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "TargetGroupAttributes": Array [
          Object {
            "Key": "deregistration_delay.timeout_seconds",
            "Value": "20",
          },
        ],
        "UnhealthyThresholdCount": 2,
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "WazuhSecurityGroup": Object {
      "Properties": Object {
        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "FromPort": 1514,
            "IpProtocol": "tcp",
            "ToPort": 1515,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "40.0.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-admin-console",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
  },
}
`;
